pkg/querier/distributor_queryable.go

```diff
@@ -36,13 +38,16 @@ type Distributor interface {
 	MetricsMetadata(ctx context.Context) ([]scrape.MetricMetadata, error)
 }
 
-func newDistributorQueryable(distributor Distributor, streamingMetdata bool, labelNamesWithMatchers bool, iteratorFn chunkIteratorFunc, queryIngestersWithin time.Duration) QueryableWithFilter {
+type partialDataEnabledFunc func(userID string) bool
```

Comment:
nit: Should the entire limits.Overrides be passed here?

Reply:
I just thought the limits.Overrides would be unnecesarily big, but I'm interested to hear others' opinions on it.

Reply:
Change made